### PR TITLE
Add Grouped Conv Backward Weight on Navi21 for ResNet50.

### PR DESCRIPTION
**Pull request by geyyer**

### CODE REVIEW

**Comment by geyyer** [comment]
@asroy, @zjing14, @poyenc, could you review the PR?

File: example/20_grouped_conv_bwd_weight/run_grouped_conv_bwd_weight_example.inc [hunk 1]
```diff
@@ -54,8 +14,17 @@ template <ck::index_t NDimSpatial>
 bool run_grouped_conv_bwd_weight(const ExecutionConfig& config,
                                  const ck::utils::conv::ConvParam& conv_param)
 {
-    constexpr ck::index_t split_k = 2;
-
+    ck::index_t split_k;
+    // Set split_k = 2 for xdl op, split_k = 1 for dl
+    if(!(ck::get_device_name() == "gfx906" || ck::get_device_name() == "gfx1030"))
```
Comment (zjing14) [inline example/20_grouped_conv_bwd_weight/run_grouped_conv_bwd_weight_example.inc:19]: Please add comments here to explain why only gfx906 and gfx1030 support split_k > 1. If it is due to a compiler issue, please add "to-do", and the corresponding jira.

Comment (geyyer) [reply example/20_grouped_conv_bwd_weight/run_grouped_conv_bwd_weight_example.inc:19]: An Xdl example had split_k = 2, but a Dl example had split_k = 1. Let me double check this inconsistency and see if we can unify the examples.

Comment (geyyer) [reply example/20_grouped_conv_bwd_weight/run_grouped_conv_bwd_weight_example.inc:21]: @zjing14, indeed, the Dl op doesn't support split_k > 1, added comments and a jira task.

**Review by zjing14 — APPROVED** [review]

**Pull request merged**